File: sync-server/src/sync/NameSyncHandler.ts
import { IdbUpdate } from './../database/IdbUpdate';
import { NetworkBuffer } from './../network/NetworkBuffer';
import { IdbUpdatePacket } from './../network/packets/IdbUpdatePacket';
import { BasePacket } from './../network/packets/BasePacket';
import { NetworkClient } from './../network/NetworkClient';
import { ISyncHandler, SyncType } from './ISyncHandler';

class NameSyncUpdateData extends IdbUpdate {
    public ptr:number;
    public name:string;
}

export class NameSyncHandler implements ISyncHandler {
    public syncType:SyncType = SyncType.Name;

    public decodePacket(updateData:NameSyncUpdateData, packet:IdbUpdatePacket) {
        updateData.ptr = packet.buffer.readUInt64();
        updateData.name = packet.buffer.readString();
    }

    public encodePacket(packet:IdbUpdatePacket, updateData:NameSyncUpdateData) {
        packet.buffer.writeUInt64(updateData.ptr);
        packet.buffer.writeString("hello_fotze");// updateData.name);
    }
}

File: sync-plugin/src/sync/NameSyncHandler.cpp
#include "sync/NameSyncHandler.h"

#include <ida.hpp>
#include <idp.hpp>
#include <name.hpp>

bool NameSyncHandler::ApplyUpdate(IdbUpdate* _updateData)
{
	auto updateData = (NameSyncUpdateData*)_updateData;
	return set_name(static_cast<ea_t>(updateData->ptr), updateData->name.c_str(), SN_NOWARN);// (packet->local ? SN_LOCAL : 0) | SN_NOWARN));
}

IdbUpdate* NameSyncHandler::DecodePacket(NetworkBufferT<BasePacket>* packet)
{
	auto updateData = new NameSyncUpdateData();
	
	// Data
	packet->Read(&updateData->ptr);

	auto str = packet->ReadString();
	if (str == nullptr)
		return nullptr;
	
	updateData->name = str;
	return updateData;
}

bool NameSyncHandler::EncodePacket(NetworkBufferT<BasePacket>* packet, IdbUpdate* _updateData)
{
	auto updateData = (NameSyncUpdateData*)_updateData;

	// Data
	packet->Write(updateData->ptr);
	packet->WriteString(updateData->name.c_str());

	return true;
}